### PR TITLE
Bump .NET SDK

### PR DESCRIPTION
**Pull request by bitsandfoxes**
Bumped the SDK and updated the `User` to `SentryUser` due to a breaking change in the mayor bump.

### CODE REVIEW

File: aspnetcore/Empower.Backend.csproj [hunk 1]
```diff
@@ -16,9 +16,13 @@
     <SentryUploadSources>true</SentryUploadSources>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)' == 'Debug'">
+    <UseSentryCLI>false</UseSentryCLI>
+  </PropertyGroup>
```
Comment (realkosty) [inline aspnetcore/Empower.Backend.csproj:21]: hm why not leave the warning?

**Review by realkosty — CHANGES_REQUESTED** [review]
Thank you for the update!!
just 1 comment

File: aspnetcore/Empower.Backend.csproj (hunk 1 above)
Comment (bitsandfoxes) [reply aspnetcore/Empower.Backend.csproj:21]: There is no `sentry-cli` auth stuff on the environment when it gets built by CodeQL and then fails horribly, breaking the build.

Comment (bitsandfoxes) [reply aspnetcore/Empower.Backend.csproj:21]: So we'd either need to find a way to set those and the auth token somehow, somewhere to be picked up by the action, which I don't know how to do, or we skip the upload for `debug` builds (for now (tm)).

Comment (realkosty) [reply aspnetcore/Empower.Backend.csproj:21]: Hm chatgpt's telling me we could set a condition like this
<PropertyGroup Condition="'$(CODEQL_ANALYSIS)' == 'true'">
    <UseSentryCLI>false</UseSentryCLI>
</PropertyGroup>

**Review by realkosty — APPROVED** [review]

File: aspnetcore/Empower.Backend.csproj (hunk 1 above)
Comment (realkosty) [reply aspnetcore/Empower.Backend.csproj:21]: Is that the same failure you had before when CodeQL was failing due to missing auth token?
 ```
 Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
  - Required package "facade/ignition" is not present in the lock file.
  - Required package "nunomaduro/collision" is not present in the lock file.
  This usually happens when composer files are incorrectly merged or the composer.json file is manually edited.
  Read more about correctly resolving merge conflicts https://getcomposer.org/doc/articles/resolving-merge-conflicts.md
  and prefer using the "require" command over editing the composer.json file directly https://getcomposer.org/doc/03-cli.md#require-r
  Error: Could not auto-detect a suitable build method
  Error: We were unable to automatically build your code. Please replace the call to the autobuild action with your custom build steps. Encountered a fatal error while running "/opt/hostedtoolcache/CodeQL/2.17.4/x64/codeql/csharp/tools/autobuild.sh". Exit code was 1 and last log line was: Error: Could not auto-detect a suitable build method. See the logs for more details.
```

Comment (realkosty) [reply aspnetcore/Empower.Backend.csproj:21]: Either way, we can figure it out later 🚢

Comment (bitsandfoxes) [reply aspnetcore/Empower.Backend.csproj:21]: Looks like ChatGPT is dreaming things up as they should be and not how they are 😅

**Pull request merged**